### DISCUSSION TITLE
Disable 'Place solution and project in the same directory' and force to true

### DISCUSSION THREAD

**Discussion by dansiegel** [dotnet/templating#7492] [General]
Is there a way with the .NET Templates to force Visual Studio to place the project and solution file in the same directory and disable the option in the UI?

**Comment by baronfel** [comment]
cc @phenning - this has been a thing we've danced around for a bit, right?

**Comment by sayedihashimi** [comment]
@dansiegel take a look at https://github.com/sayedihashimi/template-sample?tab=readme-ov-file#how-to-create-a-multi-project-solution-template, I believe that should disable that checkbox, but I'm not 100% sure.

**Reply by vernou** [reply]
I couldn't get it to work from your demo template. The option "Place solution and project in the same directory" isn't disabled and if unchecked all files (except .sln) are in subfolder.

**Reply by sayedihashimi** [reply]
@vernou can you provide a sample that is not working for you?

**Reply by vernou** [reply]
I tested with this template :
https://github.com/sayedihashimi/template-sample/tree/main/src/Content/SolutionTemplate

You can reproduce with :
```
git clone https://github.com/sayedihashimi/template-sample.git
dotnet new install template-sample\src\Content\SolutionTemplate
```
Then when you create a new project :
[vid.webm](https://github.com/user-attachments/assets/1e35c67d-1aca-468b-a8af-b6cc3801c3f7)

The option "Place solution and project in the same directory" isn't disabled.
And if unchecked all files (except .sln) are in subfolder.

**Reply by sayedihashimi** [reply]
@phenning this looks like a regression in the `Solution` template support in the New Project Dialog. The checkbox should be disabled when using that solution template. I was able to repro the issue on my machine with the latest build.

**Comment by phenning** [comment]
Shouldn’t the behavior be achieved through editorTreatAs: solution tag?

type:solution just makes us look for and use sln files in the primary outputs rather than proj files.

**Reply by sayedihashimi** [reply]
> Shouldn’t the behavior be achieved through editorTreatAs: solution tag?
> 
> type:solution just makes us look for and use sln files in the primary outputs rather than proj files.

@phenning Yeah, that was it, I guess I forgot to update that sample after we created `editorTreatAs`. I'll get the README updated soon.

@vernou I modified the `template.json` in that sample to have the following. I added `editorTreatAs`

```json
"tags": {
  "language": "C#",
  "type": "solution",
  "editorTreatAs":"solution"
},
```

After doing that things were working as expected. If you try it locally you will have to uninstall the template, make sure that VS is closed, and then restart VS. I will update the docs hopefully tomorrow. Sorry about that.

**Reply by vernou** [reply]
Idem, the option "Place solution and project in the same directory" isn't disabled.
I tested ".NET SDK 8.0.400" and "Microsoft Visual Studio Community 2022 (64-bit) - Version 17.11.4".

> If you try it locally you will have to uninstall the template, make sure that VS is closed, and then restart VS

I updated the local git repo and "editorTreatAs" is present.
Then I uninstalled the template (checked in VS it was absent) and re-installed it.
So I think the expected template version was selected in my retry.
Do I miss something?

**Reply by sayedihashimi** [reply]
@vernou I had some issues getting it to update in VS too, after uninstalling make some other change in template.json, like the name or identity, and then install again.

**Reply by vernou** [reply]
Indeed, I tested on other computer and it's worked as I expected.
The option replaced by "Create in new folder" that respect the file structure :
![image](https://github.com/user-attachments/assets/cf99dac6-9ab4-415d-893e-85695c25af38)

Thank you, your repository "template-sample" is a big help to create custom template.

**Reply by sayedihashimi** [reply]
Ok that is great. I think VS was caching the template because it was installed from a folder. I have updated the readme in the repo to mention `editorTreatAs`.
